### PR TITLE
Bug 1666167: Do a full cluster restart in case of cert redeploy

### PR DESCRIPTION
**Pull request by ewolinetz**
As part of doing a redeploy due to certs being updated we need to do a full cluster restart otherwise nodes aren't able to rejoin the cluster due to cert mismatching.

### CODE REVIEW

File: pkg/k8shandler/cluster.go [hunk 1]
```diff
@@ -389,13 +406,85 @@ func (elasticsearchRequest *ElasticsearchRequest) updateNodeStatus(status api.El
...
+
+				 // get initial cluster size
+				 size, err := GetClusterNodeCount(node.clusterName, node.self.Namespace, node.client)
+				 if err != nil {
+					 logrus.Warnf("Unable to get cluster size prior to restart for %v", node.name())
+					 return
+				 }
+
+				 // set conditions here for next check
+			 }
```
Comment (richm) [inline pkg/k8shandler/cluster.go]: formatting

File: pkg/k8shandler/cluster.go [hunk 2]
```diff
@@ -389,13 +406,85 @@ func (elasticsearchRequest *ElasticsearchRequest) updateNodeStatus(status api.El
...
 	// flush nodes
-	// call fullClusterRestart on each node
-	// update node state
+	if ok, err := DoSynchronizedFlush(node.clusterName, node.self.Namespace, node.client); !ok {
+		logrus.Warnf("Unable to perform synchronized flush: %v", err)
+	}
+
+
+	// 3 -- restart
+	// conditiong restarting true
```
Comment (richm) [inline pkg/k8shandler/cluster.go]: condition

**Comment by ewolinetz** [comment]
/hold
seeing some strangeness when restarting with statefulsets and deployments

**Pull request merged**